python/ray/data/datasource/delta_sharing_datasource.py

```diff
@@ -0,0 +1,128 @@
+        limit: Optional[int] = None,
+        version: Optional[int] = None,
+        timestamp: Optional[str] = None,
+    ):
+        self._table = table
+        self._rest_client = rest_client
+        self._jsonPredicateHints = jsonPredicateHints
+
+        if limit is not None:
+            assert (
+                isinstance(limit, int) and limit >= 0
+            ), "'limit' must be a non-negative int"
+
+        self._limit = limit
+        self._version = version
+        self._timestamp = timestamp
+
+        self._response = self._rest_client.list_files_in_table(
+            table,
+            jsonPredicateHints=self._jsonPredicateHints,
+            limitHint=self._limit,
+            version=self._version,
+            timestamp=self._timestamp,
+        )
+
+        if len(self._response.add_files) == 0 or self._limit == 0:
+            logger.warning("No files found from the delta sharing table or limit is 0")
+
+        schema_json = loads(self._response.metadata.schema_string)
+        self._converters = to_converters(schema_json)
+
+    def estimate_inmemory_data_size(self) -> Optional[int]:
+        return None
+
+    def get_read_tasks(self, parallelism: int) -> List[ReadTask]:
+        assert parallelism > 0, f"Invalid parallelism {parallelism}"
+
+        read_tasks = []
+        # get file list to be read in this task and preserve original chunk order
+        for files in np.array_split(self._response.add_files, parallelism):
+            files = files.tolist()
+            metadata = BlockMetadata(
+                num_rows=None,
+                schema=None,
+                input_files=files,
+                size_bytes=None,
+                exec_stats=None,
+            )
+            converters = self._converters
+            read_task = ReadTask(
+                lambda f=files: _read_files(f, converters),
+                metadata,
+            )
+            read_tasks.append(read_task)
+
+        return read_tasks
+
+
+def _read_files(files, converters):
+    """Read files with Delta Sharing."""
+    for file in files:
+        yield DeltaSharingReader._to_pandas(
+            action=file, converters=converters, for_cdf=False, limit=None
+        )
+
+
+def setup_delta_sharing_connections(url: str) -> DataSharingRestClient:
+    """
+    Set up delta sharing connections based on the url.
+
+    :param url: a url under the format "<profile>#<share>.<schema>.<table>"
+    :
+    """
+    from delta_sharing.protocol import DeltaSharingProfile, Table
+
+    profile_str, share, schema, table_str = _parse_delta_sharing_url(url)
+    table = Table(name=table_str, share=share, schema=schema)
+
+    profile = DeltaSharingProfile.read_from_file(profile_str)
+    rest_client = DataSharingRestClient(profile)
+    return table, rest_client
+
+
+def _parse_delta_sharing_url(url: str) -> Tuple[str, str, str, str]:
+    """
+    Developed from delta_sharing's _parse_url function.
```

Comment:
Any reason why we cannot reuse the original function? Also let's add a link to the original function so people can reference.

Reply:
I think in general we do not want to import a private function from another module.
I can add the link here.